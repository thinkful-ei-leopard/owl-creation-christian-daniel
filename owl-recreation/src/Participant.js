import React from 'react';
import participants from './STORE';

function Participant(participant) {
    return (
        <div className='Participant'>
            <img alt='participant avatar' src={participant.avatar} />
            <p>{participant.name}</p>
            <div className={participant.inSession ? 'inSession' : 'notInSession'}></div>
            <div className={participant.onStage ? 'onStage' : 'notOnStage'}></div>
        </div>
    )
}

export default Participant;