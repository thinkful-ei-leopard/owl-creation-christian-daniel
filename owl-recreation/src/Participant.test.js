import React from 'react';
import ReactDOM from 'react-dom';
import Participant from './Participant';
import renderer from 'react-test-renderer';

describe('<Participant />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Participant />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<Participant id='2' name='Caty Flucker' avatar='https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1' inSession='true' onStage='false' />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});