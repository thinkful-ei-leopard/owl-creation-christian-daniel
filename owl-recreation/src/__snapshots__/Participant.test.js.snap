// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Participant /> renders the UI as expected 1`] = `
<div
  className="Participant"
>
  <img
    alt="participant avatar"
    src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
  />
  <p>
    Caty Flucker
  </p>
  <div
    className="inSession"
  />
  <div
    className="onStage"
  />
</div>
`;
